runelite-client/src/main/java/net/runelite/client/RuneLite.java

```diff
@@ -223,8 +223,22 @@ public ClientUpdateCheckMode convert(String v)
 
 		if (options.has("debug"))
 		{
-			final Logger logger = (Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
-			logger.setLevel(Level.DEBUG);
+			// during plugin-hub dev, not all devs use logback-classic
+			// make sure we have logback-classic before attempting to set level
+			// setting level at runtime is not supported by the slf4j-api alone
+			final org.slf4j.Logger rootLogger = LoggerFactory.getLogger(org.slf4j.Logger.ROOT_LOGGER_NAME);
+			if (StaticLoggerBinder.getSingleton().getLoggerFactoryClassStr().equals("ch.qos.logback.classic.util.ContextSelectorStaticBinder"))
```

Comment:
Can't this just be an `instanceof Logger` check?

Reply:
> Can't this just be an `instanceof Logger` check?

Unfortunately not. I anticipated that an instanceof check should work, since the instanceof operator does not actually cause the ClassLoader to load in the right-hand operand class, but apparently it does still require that the ClassLoader can *find* the class. Therefore, when I tested it with an instanceof check, it still threw the ClassNotFoundEx.

For context on how I determined that the information was in StaticLoggerBinder, that is the same class SLF4J uses to determine which binding is in the classpath. This is also the same class that will throw warnings if there are multiple bindings on the classpath. It provides `getLoggerFactoryClass()` to get the `Class<?>` instance, or `getLoggerFactoryClassStr()` which is exclusively the class name.